src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -972,6 +972,8 @@ private List<ChainedHeader> CreateNewHeaders(List<BlockHeader> headers)
             var newChainedHeaders = new List<ChainedHeader>();
 
             ChainedHeader newChainedHeader = this.CreateAndValidateNewChainedHeader(headers[newHeaderIndex], previousChainedHeader);
+            if (newChainedHeader == null) return newChainedHeaders;
```

Comment:
Trace logs are needed here no?

Also better just use the bottom of the method as exit point so put an if here I would suggest